spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfigurationTests.java

```diff
@@ -139,6 +142,25 @@ public void eventListenersAreAutoRegistered() {
 				.onPreSave(any(Event.class));
 	}
 
+	@Test
+	public void autoIndexConfiguration() {
+		load(CustomConfigurationFactory.class);
+		assertThat(this.context.getBean(org.neo4j.ogm.config.Configuration.class)
+				.getAutoIndex()).isEqualTo(AutoIndexMode.NONE);
+
+		load(CustomConfigurationFactory.class, "spring.data.neo4j.auto-index=assert");
```

Comment:
I don't understand the purpose of this. Are you trying to test that the binder is able to properly map that value? If that's the goal, then you're testing the binder, not your code.

I'd have two tests. One that validates that the mode is none if you don't do anything. And one that overrides the value to one of the other values. But you should do that in separates tests (load does not close the context you've created previously).

It would be much better to do something and validate that it has the expected impact. But you can't do that if neo4j isn't available.